src/agent0/core/hyperdrive/interactive/i_hyperdrive.py

```diff
@@ -245,7 +244,7 @@ def _open_long(self, agent: HyperdriveAgent, base: FixedPoint) -> OpenLong:
         agent.policy.set_next_action(HyperdriveActionType.OPEN_LONG, base)
         # TODO expose async here to the caller eventually
         trade_results: list[TradeResult] = asyncio.run(
-            async_execute_agent_trades(self.interface, [agent], liquidate=False, interactive_mode=True)
+            async_execute_agent_trades(self.interface, agent, liquidate=False, interactive_mode=True)
```

Comment:
Again, it looks like this function was split up into 3 different places, we have `async_execute_agent_trades`, `async_execute_multi_agent_trades`, and `async_execute_single_agent_trades`. I'd rather have one general purpose function that's being called everywhere instead of copies of functions being called in different places.